src/Components/Web/src/Forms/InputFile/InputFileJsCallbacksRelay.cs

```diff
@@ -13,6 +14,7 @@ internal class InputFileJsCallbacksRelay : IDisposable
 
         public IDisposable DotNetReference { get; }
 
+        [DynamicDependency(nameof(NotifyChange))]
```

Comment:
This is subtle. Having to reason about all the possible code paths that could result in a given `[JSInvokable]` method to be used. I don't have a better suggestion than what you've done here, but it something we'll have to think hard about in all code reviews that could impact the possible code paths that lead to a JS invokable member being used.

Reply:
You're right. I think `JSInvokable` is one of the few feature that currently has no way to signal a linker warning and you would only find out at runtime. Makes @javiercn's [issue to test these as part of our builds](https://github.com/dotnet/aspnetcore/issues/29903) doubly important.